### PR TITLE
WIP accountant billing page

### PR DESCRIPTION
**Pull request by tingaloo**
I rule and have completed some work on Case Manager that's ready for review!

Have a basic accounting view that shows patients with fulfillments. It paginates by 25. 

BUG: On fulfillment update, if you previously had a search filter, it gets reset and loads the entire patient list again. Look very closely between 2nd and 3rd screenshot, on any update (in this case check number), it will reset to original set of patients without search filter.

TODOs
- [x] Implement pagination
- [x] Fulfillment partial on edit
- [x] Write tests
- [x] Search Functionality
- [ ] Additional ajax tests
- [ ] Possibly fix search bug on fulfillment update
- [x] Minor style changes

This pull request makes the following changes:
* Add table
* Add sorting 

screen shot 2017-10-13 at 10 19 22 am
screen shot 2017-10-13 at 10 19 39 am
screen shot 2017-10-13 at 10 19 59 am

It relates to the following issue #s: 
* Bumps #858

### CODE REVIEW

**Comment by colinxfleming** [comment]
@tingaloo I think this is pretty much a perfect use case for `will_paginate`!

**Comment by colinxfleming** [comment]
I think this is 100% on the right track! Thanks for picking it up @tingaloo !

**Comment by tingaloo** [comment]
Soo close getting this to work.

For tests, i can't use ```wait_for_ajax``` because phantom driver is missing?

And I discovered creating a form within a table is not allowed https://stackoverflow.com/questions/5967564/form-inside-a-table. So having the inline edits within the table cell won't work unless we wrap tables with ```form```. That means we have something that looks like so if we want inline editing to technically work. Otherwise the form will never wrap properly and no inline...

screen shot 2017-10-11 at 8 31 03 am

As of now, the form div looks like so with no children.
```<form role="form" class="edit_patient" id="edit_patient_59d6e4b52cde1127ba38f255" action="/patients/59d6e4b52cde1127ba38f255" accept-charset="UTF-8" data-remote="true" method="post"></form>```

**Comment by colinxfleming** [comment]
`wait_for_ajax` phantom error is a mystery to me -- is that in CI or in docker or your local? 

I didn't know that about forms inside `tr`s! Sorry for the scrub directions on my part. I think we have a few options here, and you're free to go with your favorite:

a) style the form so that it's stylistically identical to a table row
b) instead of the form, have a link in the `tr` that pops up a modal, which populates it with `_fulfillment` for that patient (this might be nice because it'd let us reuse the partial)
c) something else tbd?

File: Gemfile [hunk 1]
```diff
@@ -34,6 +34,9 @@ gem 'rack-test', '~> 0.6.3', require: 'rack/test'
 gem 'clinic_finder', '~> 0.0.1'
 gem 'geokit'
 gem 'secure_headers', '~> 3.6', '>= 3.6.4'
+gem 'kaminari-mongoid'
+gem 'kaminari'
```
Comment (colinxfleming) [inline Gemfile]: Haven't heard of this lib but it looks sweet! 👍

File: app/controllers/accountants_controller.rb [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+  
+  def index
+    @patients = Patient.where('fulfillment.fulfilled' => true)
+    @patients = Kaminari.paginate_array(@patients).page(params[:page]).per(25)
+  end
 
   def search
-    @results = Patient.search params[:search]
-
+    @results = Patient.where('fulfillment.fulfilled' => true).where(:name => /#{params[:search]}/i)
```
Comment (colinxfleming) [inline app/controllers/accountants_controller.rb]: I'd suggest taking advantage of the `Patient.search` method here and chaining it against the `where`. e.g. `Patient.search(params[:search]).where('fulfillment.fulfilled' => true)` if memory serves. (s/o to @KatSDC for writing a slamming search algo!)

File: app/helpers/accountants_helper.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Functions primarily related to accountant billing view
+module AccountantsHelper
+
+  def patientInitials(patient)
```
Comment (colinxfleming) [inline app/helpers/accountants_helper.rb]: Might behoove us to move this to the `Patient` object as an instance method, so we can do `@patient.initials`? thoughts?

File: app/controllers/accountants_controller.rb (hunk 2 above)
Comment (tingaloo) [reply app/controllers/accountants_controller.rb]: Can you tell me where that method definition is handled, i'd like to see if it supports cases.

File: Gemfile (hunk 1 above)
Comment (tingaloo) [reply Gemfile]: I think ```will_paginate``` didn't support ```mongoid```, this looks cooler and provides cool pagination links.

**Comment by tingaloo** [comment]
```wait_for_ajax``` error is from local. 

I'll probably do option b unless we REALLY want inline editing within a table. I'll close this PR up and start a new issue for that.

**Comment by colinxfleming** [comment]
I think you can probably keep this PR cooking, right? Looks like you have most of the backend machinery in place already, and what's left is mostly just beating on views, I'd guess?

File: app/controllers/accountants_controller.rb (hunk 2 above)
Comment (colinxfleming) [reply app/controllers/accountants_controller.rb]: The search internals for patients are here -- https://github.com/DCAFEngineering/dcaf_case_management/blob/master/app/models/concerns/patient_searchable.rb

**Comment by colinxfleming** [comment]
cc @eheintzelman -- does that strategy (pop up modal rather than inline form) sound ok to you? Anything special we should do around behavior?

**Comment by eheintzelman** [comment]
@colinxfleming That should work.  I'm imagining a pencil icon to the right of each row, indicating the data is editable. The pencil icon triggers a modal with all the fields. Something similar to what you see in LinkedIn when you edit a section of your profile.

**Comment by colinxfleming** [comment]
@tingaloo anything else on this front?

**Comment by tingaloo** [comment]
Ready to go. Bugs are minor and we can spin them off as individual issues.

**Comment by colinxfleming** [comment]
Gonna in-house this so we can all push to it -- closing in favor of #1292

**Pull request closed without merging**